widgets/index-selector.js

```diff
@@ -37,17 +39,22 @@ function indexSelector({
       }
     },
 
+    setIndex: function(helper, indexName) {
+      helper.setIndex(indexName).search();
```

Comment:
:+1: we had a few places where we were doing the .search inside the component. Much better to do it here